File: base_tools/src/main/java/qzl/com/tools/operate/java/ReadProperties.java
package qzl.com.tools.operate.java;


import java.io.InputStream;
import java.util.Properties;

public class ReadProperties {
    private static Properties defaultProperty;
    private static Properties configProperty;
    static {
		initConfig();
    }

	private static void initConfig() {
		configProperty = new Properties();
		InputStream stream =
				ReadProperties.class.getResourceAsStream("/assets/config.properties");
		try {
			configProperty.load(stream);
		} catch (Exception e) {

		}
		configProperty.putAll(System.getProperties());
		if (getConfigPropertyByBoolean("isRelease")){
			//如果是发布版本
			init("release");
		}else {
			//如果是调试版本
			init("debug");
		}
	}

	/**
	 * @desc 配置参数的名称
	 * @author 强周亮
	 * @time 2019-01-24 11:51
	 */
	static void init(String str) {
        defaultProperty = new Properties();
        InputStream stream =
        		ReadProperties.class.getResourceAsStream("/assets/"+str+".properties");
		try {
			defaultProperty.load(stream);
		} catch (Exception e) {

		}
		defaultProperty.putAll(System.getProperties());
    }
	/**
	 * @param propertyName :对象名称
	 * @return 对象值
	 * 取得配置文件的对象值
	 */
	private static Boolean getConfigPropertyByBoolean(String propertyName) {
		return Boolean.valueOf(String.valueOf(ReadProperties.configProperty.get(propertyName)));
	}
	/**
	 * @param propertyName :对象名称
	 * @return 对象值
	 * 取得配置文件的对象值-公共config里面的属性值
	 */
	public static String getConfigPropertyByStr(String propertyName) {
		return String.valueOf(ReadProperties.configProperty.get(propertyName));
	}
    /**
	 * @param propertyName :对象名称
	 * @return 对象值
	 * 取得配置文件的对象值
	 */
	public static String getPropertyByStr(String propertyName) {
		return String.valueOf(ReadProperties.defaultProperty.get(propertyName));
	}
	/**
	 * @param propertyName :对象名称
	 * @return 对象值
	 * 取得配置文件的对象值
	 */
	public static int getPropertyByInt(String propertyName) {
		return Integer.parseInt(ReadProperties.getPropertyByStr(propertyName));
	}
   
}

File: QGMultipleModelFrame/base_tools/src/main/java/qzl/com/tools/operate/ReadProperties.kt
package qzl.com.tools.operate

import android.content.Context
import com.qzl.toast.MyToast
import qzl.com.tools.utils.AppUtil.isApkInDebug
import java.util.*

object ReadProperties {
    private var defaultProperty: Properties? = null
    fun setContext(context: Context?) {
        if (context == null) {
            MyToast.showLong("请先初始化配置文件读取类！！！")
            return
        }
        if (isApkInDebug(context)) {
            init("debug")
        } else {
            init("release")
        }
    }

    /**
     * @desc 配置参数的名称
     * @author 强周亮
     * @time 2019-01-24 11:51
     */
    fun init(str: String) {
        if (defaultProperty == null) {
            defaultProperty = Properties()
        }
        val stream =
            ReadProperties::class.java.getResourceAsStream("/assets/$str.properties")
        try {
            defaultProperty!!.load(stream)
        } catch (e: Exception) {
        }
        defaultProperty!!.putAll(System.getProperties())
    }

    /**
     * @param propertyName :对象名称
     * @return 对象值
     * 取得配置文件的对象值
     */
    fun getPropertyByStr(propertyName: String): String {
        return defaultProperty!![propertyName].toString()
    }

    /**
     * @param propertyName :对象名称
     * @return 对象值
     * 取得配置文件的对象值
     */
    fun getPropertyByInt(propertyName: String): Int {
        return getPropertyByStr(propertyName).toInt()
    }
}